tests/resources/test_attachment.py

```diff
@@ -30,6 +31,14 @@ def test_1_add_remove_attachment_using_filestream(self):
 
     def test_2_add_remove_attachment_using_filename(self):
         issue = self.jira.issue(self.issue_1)
+        attachment_no_filename_specified = self.jira.add_attachment(
+            issue,
+            TEST_ATTACH_PATH,
+        )
+        new_attachment = self.jira.attachment(attachment_no_filename_specified.id)
+        msg = f"attachment, no filename specified {new_attachment.__dict__} of issue {issue}"
+        self.assertEqual(new_attachment.filename, Path(TEST_ATTACH_PATH).name, msg=msg)
+        #
```

Comment:
excellent test case, however, let's put this into a new method, the test name doesn't make sense here

We don't need to add a number to the tests so just something like:


```suggestion
    def test_add_attachment_without_filename(self):
        issue = self.jira.issue(self.issue_1)
        attachment_no_filename_specified = self.jira.add_attachment(
            issue=issue,
            attachment=TEST_ATTACH_PATH,
            filename=None,
        )
        new_attachment = self.jira.attachment(attachment_no_filename_specified.id)
        msg = f"attachment, no filename specified {new_attachment.__dict__} of issue {issue}"
        assert new_attachment.filename == Path(TEST_ATTACH_PATH).name, msg

    def test_2_add_remove_attachment_using_filename(self):
        issue = self.jira.issue(self.issue_1)
```

(I've also moved away from the self.assert methods, one day we will clean out the legacy unittest syntax)